src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DefaultProjectSnapshotManager.cs

```diff
@@ -149,10 +149,15 @@ public override void ProjectUpdated(ProjectSnapshotUpdateContext update)
 
```

Comment:
Above here, we're checking if the snapshot is dirty and if so to re-notify background workers. Should we be notifying anyone? Seems like we'd be notifying not up-to-date information.

Reply:
I think you might be mixing this up with `NotifyListeners`. We need to `NotifyBackgroundWorker` if the snapshot is dirty so we can compute up to date values.